File: DSA/LEETCODE_AUTOMATED/0069-sqrtx/solution.java
class Solution {
    public int mySqrt(int x) {
        int l=1,r=x,mid=0;
        while(l<=r){
            mid=l+(r-l)/2;
            if((long)mid*mid > (long)x){
                r=mid-1;
            }else if((long)mid*mid < (long)x){
                l=mid+1;
            }else{
                return mid;
            }
        }
        return Math.round(r);
    }
}

File: DSA/LEETCODE_AUTOMATED/0069-sqrtx/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l,r=1,x
        while l<=r:
            m=(l+r)//2
            m_sqr=m*m
            if m_sqr == x:
                return m
            if m_sqr < x:
                l=m+1
            if m_sqr > x:
                r=m-1
        return r
